src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -347,6 +357,10 @@ public unsafe IntPtr GetOrCreateComInterfaceForObject(object instance, CreateCom
         private unsafe ManagedObjectWrapper* CreateCCW(object instance, CreateComInterfaceFlags flags)
         {
             ComInterfaceEntry* userDefined = ComputeVtables(instance, flags, out int userDefinedCount);
+            if (userDefinedCount < 0)
+            {
+                throw new ArgumentException();
```

Comment:
Does CoreCLR throw general ArgumentException in this case, or is there a more specific error message?

Reply:
It returns `E_INVALIDARG` which then call `COMPlusThrowHR(hr);` so I think yes, that's proper response.
https://github.com/dotnet/runtime/blob/c02d6dac78532a841ae9dcebe621b9e65ebc2920/src/coreclr/interop/interoplib.cpp#L33